controllers/keda/triggerauthentication_controller.go

```diff
@@ -116,3 +121,80 @@ func (r *TriggerAuthenticationReconciler) UpdatePromMetricsOnDelete(namespacedNa
 
 	delete(triggerAuthPromMetricsMap, namespacedName)
 }
+
+func GetTriggerAuth(ctx context.Context, client client.Client, triggerAuthRef *kedav1alpha1.ScaledObjectAuthRef, namespace string) (client.Object, error) {
+	if triggerAuthRef == nil {
+		return nil, fmt.Errorf("triggerAuthRef is nil")
+	}
+
+	if triggerAuthRef.Kind == "" || triggerAuthRef.Kind == "TriggerAuthentication" {
+		triggerAuth := &kedav1alpha1.TriggerAuthentication{}
+		err := client.Get(ctx, types.NamespacedName{Name: triggerAuthRef.Name, Namespace: namespace}, triggerAuth)
+		if err != nil {
+			return nil, err
+		}
+		return triggerAuth, nil
+	} else if triggerAuthRef.Kind == "ClusterTriggerAuthentication" {
+		clusterTriggerAuth := &kedav1alpha1.ClusterTriggerAuthentication{}
+		err := client.Get(ctx, types.NamespacedName{Name: triggerAuthRef.Name, Namespace: namespace}, clusterTriggerAuth)
+		if err != nil {
+			return nil, err
+		}
+		return clusterTriggerAuth, nil
+	}
+	return nil, fmt.Errorf("unknown trigger auth kind %s", triggerAuthRef.Kind)
+}
+
+func GetTriggerAuthStatus(triggerAuth client.Object) (*kedav1alpha1.TriggerAuthenticationStatus, error) {
+	switch obj := triggerAuth.(type) {
+	case *kedav1alpha1.TriggerAuthentication:
+		return &obj.Status, nil
+	case *kedav1alpha1.ClusterTriggerAuthentication:
+		return &obj.Status, nil
+	default:
+		return nil, fmt.Errorf("unknown trigger auth status %s", obj)
+	}
+}
+
+func UpdateTriggerAuthenticationStatusHandler(ctx context.Context, logger logr.Logger, client client.Client, namespace string, scaleTriggers []kedav1alpha1.ScaleTriggers, statusHandler func(*kedav1alpha1.TriggerAuthenticationStatus) *kedav1alpha1.TriggerAuthenticationStatus) (string, error) {
```

Comment:
I don't think this should be in the TA_controller code, as it is not used here at all. I think that it would be better to create a new package `pkg/status` and move this logic there. (also move there stuff from `pkg/util/status.go`)